bounce.py

```
@@ -23,21 +22,82 @@
+
+
+def is_rate_limited(ip: str) -> bool:
+    """Check if the given IP is rate-limited."""
+    if not rate_limit_enabled:
+        return False
+
+    if ip not in rate_limit_store:
+        rate_limit_store[ip] = []
+
+    request_times = rate_limit_store[ip]
+    current_time = time.time()
```

Ruff

Undefined name `time`

(F821)